src/ProductView/ProductView.js

```diff
@@ -103,6 +103,13 @@ class ProductView extends Component {
 
     onChangeFilterInput(currentFilterId, value) {
         const updatedFilters = this.state.filter;
+
+        if (!value) {
+            delete updatedFilters[currentFilterId];
```

Comment:
From my point of view it looks like a smart solution.
Nevertheless, It would be great to delete any filter by clicking on a small cross (x).